Makefile

```diff
@@ -1,122 +1,92 @@
-tag := latest
-git_hash ?= $(shell git log -1 --pretty=format:%h)
-
-base_dir ?= $(shell git rev-parse --show-toplevel)
-examples ?=  ../$(base_dir)/examples
-app_dir ?= $(shell git rev-parse --show-prefix)
-
-CONDA_EXE := conda
-CONDA_ENV ?= $(base_dir)/env
-env := $(CONDA_ENV)
-conda_run := $(CONDA_EXE) run -p $(env)
-PYTEST_EXE := $(CONDA_ENV)/bin/pytest
-
 MIN_NODE_VER := 14
-MIN_NPM_VER := 6
-NODE_VER    := $(shell node -v | cut -d. -f1 | sed 's/^v\(.*\)/\1/')
-NPM_VER     := $(shell npm -v | cut -d. -f1)
-
-ifeq ($(shell uname -s), Darwin)
-    SED_I_ARG := -i ''
-else
-    SED_I_ARG := -i
-endif
+MIN_NPM_VER  := 6
+MIN_PY3_VER  := 8
+NODE_VER     := $(shell node -v | cut -d. -f1 | sed 's/^v\(.*\)/\1/')
+NPM_VER      := $(shell npm -v | cut -d. -f1)
+PY3_VER      := $(shell python3 -c "import sys;t='{v[1]}'.format(v=list(sys.version_info[:2]));print(t)")
+PY_OK        := $(shell python3 -c "print(int($(PY3_VER) >= $(MIN_PY3_VER)))")
+
+all:
+	@echo "\nThere is no default Makefile target right now. Try:\n"
+	@echo "make setup - check your environment and install the dependencies."
+	@echo "make clean - clean up auto-generated assets."
+	@echo "make build - build PyScript."
+	@echo "make precommit-check - run the precommit checks (run eslint)."
+	@echo "make test-integration - run all integration tests sequentially."
+	@echo "make fmt - format the code."
+	@echo "make fmt-check - check the code formatting.\n"
 
 .PHONY: check-node
 check-node:
 	@if [ $(NODE_VER) -lt $(MIN_NODE_VER) ]; then \
-		echo "Build requires Node $(MIN_NODE_VER).x or higher: $(NODE_VER) detected"; \
+		echo "\033[0;31mBuild requires Node $(MIN_NODE_VER).x or higher: $(NODE_VER) detected.\033[0m"; \
```

Comment:
I *think* you need the `-e` flag to allow colors or other modifiers in echo, at least on Linux.

Reply:
Works for me (on Linux) without `-e` flag.

Reply:
some explanation and example in here https://linuxsimply.com/bash-scripting-tutorial/input-output/output/echo-command/ (first thing I've found, really) but maybe if you `$(tput ...)` that's not needed ... my point was rather that usually for special things `-e` doesn't hurt.

Reply:
:+1: OK... so just adding `-e` won't hurt. FWIW - I'm on a vanilla Ubuntu LTS.

Reply:
OK... so this is frustrating: in a `Makefile` it appears that `echo` will echo the flags... so I see a spare `-e` prepended to the messages. :frowning_face: 

Google-fu isn't helping either... so I'm gonna just avoid using `-e`.

Reply:
if this is a Makefile specific thing then I am OK, but in my Ubuntu I have a few shenanigans instead:

![Screenshot 2023-10-24 102646](https://github.com/pyscript/pyscript/assets/85749/ea77f047-b747-4ba9-92ae-235492f434aa)